src/services/AppSettings/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useObservableState } from 'observable-hooks'
+import { BehaviorSubject } from 'rxjs'
+import { Dispatch, SetStateAction } from 'react'
+import { appSettings } from '.'
+
+function createPropSetter<T>(prop: BehaviorSubject<T>): Dispatch<SetStateAction<T>> {
+  function isUpdater(value: unknown): value is (prevState: T) => T {
+    return typeof value === 'function'
+  }
+
+  return (action: T | ((prevState: T) => T)) => {
+    if (isUpdater(action)) {
+      prop.next(action(prop.value))
+    } else {
+      prop.next(action)
+    }
+  }
+}
+
+export function useDefaultLanguage() {
+  return [useObservableState(appSettings.defaultLanguage$), createPropSetter(appSettings.defaultLanguage$)] as const
```

Comment:
Actually, I would suggest switching languages by URL as follows

1. explorer.nervos.org/page => 'en'
2. explorer.nervos.org/zh/ => 'zh'
3. explorer.nervos.org/blah => redirect to explorer.nervos.org

It would be much simpler and user-friendly because they can visit page in the specific language

Reply:
I haven't actually researched the cost of implementing this under the current architecture. We need to create a new issue to handle this.

Reply:
https://github.com/Magickbase/ckb-explorer-public-issues/issues/466